src/main/java/com/endava/internship/collections/Student.java

```diff
@@ -1,32 +1,57 @@
 package com.endava.internship.collections;
 
 import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.Objects;
 
-/**
- * The class that defines the element that will be contained by your collection
- */
-public class Student //TODO consider implementing any interfaces necessary for your collection
-{
-    private String name;
-    private LocalDate dateOfBirth;
-    private String details;
+public class Student {
+    private final String name;
+    private final LocalDate dateOfBirth;
+    private final String details;
 
-    public Student(String name, LocalDate dateOfBirth, String details) {
+    public Student (String name, LocalDate dateOfBirth, String details) {
         this.name = name;
         this.dateOfBirth = dateOfBirth;
         this.details = details;
     }
 
-    public String getName() { return name; }
+    public String getName () {
+        return name;
+    }
+
+    public LocalDate getDateOfBirth () {
+        return dateOfBirth;
+    }
+
+    public String getDetails () {
+        return details;
+    }
+
+    public int compareTo (Student student) {
```

Comment:
If you want the student to be comparable, then this class can implement Comparable and method compareTo() will be overridden from that interface.